lints/lint_dnsname_contains_empty_label.go

```diff
@@ -0,0 +1,52 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"strings"
+)
+
+type DNSNameEmptyLabel struct {
+	// Internal data here
+}
+
+func (l *DNSNameEmptyLabel) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameEmptyLabel) CheckApplies(c *x509.Certificate) bool {
+	return true
+}
+
+func domainHasEmptyLabel(domain string) bool {
+	labels := strings.Split(domain, ".")
+	for _, elem := range labels {
+		if elem == "" {
+			return true
+		}
+	}
+	return false
+}
+
+func (l *DNSNameEmptyLabel) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	result := ResultStruct{Result: Pass}
+	if domainHasEmptyLabel(c.Subject.CommonName) {
```

Comment:
We probably don't want to be running this on CN if this isn't a subscriber certificate.